### DISCUSSION TITLE
Consider an additional license for commercial use?

### DISCUSSION THREAD

**Discussion by catty3817** [SilentVoid13/Templater#1323] [Ideas & Feature Requests]
I've recently become an Obsidian convert and use it daily at my employer. I've got ADHD that has been further complicated by Long COVID effects and Obsidian has changed my life as a second, more reliable memory.

Unfortunately, my employer will not allow any use of AGPL software. Any open source software we use, must first be checked into our internal source repos to help reduce supply chain risks. Checking an AGPL project into our internal repos might inadvertantly get mixed into a future product or even be interpreted as "derived from" in a legal challenge if we were just using it in Obsidian templates. Right, wrong, or indifferent, lawyers are gonna lawyers.

So, what I'm asking is for a commercial license, akin to the Obsidian Commercial License. You could charge for it one time, annually, or even something like "Template commercial license applies if Obsidian has an active Commercial license". The latter doesn't offer compensation, but eliminates overhead and protects anonymity of the author of Template. 

To be clear, this doesn't address the topic I've seen online of "how do I make a profitable Obsidian plugin?" It merely provides an option for limited use (not derived works) for situations where the freedom provided by AGPL isn't needed and requirements of the same are not permissable. 

I know my employer isn't the only one that enforces this limitation as my previous employer that largely produces software under the Apache license had the same requirements.

As it stands, my option is to forgo using Templater on my work devices and just rely on the inbuilt limited template support, or write my own template plugin and release it under a more permissive license.

**Comment by Zachatoo** [comment]
Not sure if they'll respond, but I believe @SilentVoid13 will have to make a decision here, since the license is listed in their name.

You may need to reach out to them on another communication channel, such as email or Discord.

**Comment by SilentVoid13** [comment]
Hi @catty3817  that's an interesting problem. However, I fail to understand how Templater can represent a supply chain risk as this is not a library or anything of that sort that could end up in production code. 

As for an "Obsidian Commercial License"-style, I'm not really a fan of it either. Templater has always been a free and open-source tool and I don't plan on commercializing it.

I would also prefer to keep the license as is as well. I like the protections offered by the AGPL. If this is a problem faced by a lot of users besides you, I'm willing to consider it though.

**Reply by catty3817** [reply]
I completely understand that stance. To clarify, I'm not at all suggesting you get rid of the AGPL, but suggesting an optional dual license scenario that might appease lawyers. For example [Mattermost](https://github.com/mattermost/mattermost/blob/master/LICENSE.txt) offers MIT licensing for the compiled Typescript and Golang code and further states that they won't pursue copy left enforcement unless it's a "modified version" or "work based on" Mattermost as defined in the AGPL. 

This article, "[AGPL is a non-starter for most companies](https://opencoreventures.com/blog/2023-10-agpl-license-is-a-non-starter-for-most-companies/)", I think describes the concern, but goes on to clarify the point you made about application software vs a library. I think the lawyer stance is that they don't know what the limits would be in a court challenge.

Anyway, thanks for responding. Like you said, if others run into the issue, maybe it's worth the effort. I can imagine how this might be coming across and I assure you I'm not trying to snake around open source community..my closest friends would call me an open source zealot and wouldn't believe I was having this discussion.

Thanks again for the time.
